src/common/stream_coder.h

```diff
@@ -0,0 +1,23 @@
+#ifndef STREAM_CODER_H
+#define STREAM_CODER_H
+
+#include "allocator.h"
+#include <nan.h>
+
+using namespace v8;
+
+class StreamCoder : public Nan::ObjectWrap {
+  public:
+    const uint8_t* next_in;
+    size_t available_in;
```

Comment:
Cool, looks like that would be usable, yep. Might be nice to do that separately, though, there are already enough changes in this PR. :)